Buildings/Fluid/Storage/Plant/TankBranch.mo

```diff
@@ -0,0 +1,316 @@
+within Buildings.Fluid.Storage.Plant;
+model TankBranch
+  "(Draft) Model of the tank branch where the tank can potentially be charged remotely"
+
+  replaceable package Medium =
+    Modelica.Media.Interfaces.PartialMedium "Medium package";
+
+  parameter Boolean allowRemoteCharging = true
+    "= true if the tank is allowed to be charged by a remote source";
+
+  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal(min=0)
+    "Nominal mass flow rate";
+  parameter Modelica.Units.SI.MassFlowRate mTan_flow_nominal(min=0)
+    "Nominal mass flow rate for CHW tank branch";
+  parameter Modelica.Units.SI.PressureDifference dp_nominal
+    "Nominal pressure difference";
+  parameter Modelica.Units.SI.Temperature T_CHWS_nominal=7+273.15
+    "Nominal temperature of CHW supply";
+  parameter Modelica.Units.SI.Temperature T_CHWR_nominal=12+273.15
+    "Nominal temperature of CHW return";
+
+  Buildings.Fluid.FixedResistances.PressureDrop preDroTan(
+    redeclare package Medium = Medium,
+    final allowFlowReversal=true,
+    final m_flow_nominal=mTan_flow_nominal) "Flow resistance on tank branch"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={50,30})));
+  Modelica.Blocks.Interfaces.RealOutput mTan_flow
+    "Mass flow rate through the tank" annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={-30,110}),  iconTransformation(
+        extent={{10,-10},{-10,10}},
+        rotation=180,
+        origin={110,20})));
+  Buildings.Fluid.Movers.SpeedControlled_y pum(
+    redeclare package Medium = Medium,
+    per(pressure(dp=dp_nominal*{2,1.2,0}, V_flow=(m_flow_nominal)/1.2*{0,1.2,2})),
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    allowFlowReversal=true,
+    addPowerToMedium=false,
+    y_start=0,
+    T_start=T_CHWR_nominal) "Secondary CHW pump" annotation (Placement(
+        transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={-70,-40})));
+
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage valDis(
```

Comment:
Maybe I missed a previous conversation, but what is the purpose of this valve, given that the pump speed is controlled? And why is it only relevant when remote charging is allowed? From the controller, it seems like the valve is closed when the pump is off, and fully open otherwise, which seems redundant. The check valve already makes sure that there is no back flow.